### ISSUE TITLE
Writing some of the post's metadata to the file properties

### ISSUE THREAD

**Issue by RiabikovR** [Bionus/imgbrd-grabber#1802]
**Is your feature request related to a problem? Please describe**

A lot of images on Pixiv, and boorus, that primarily gather their content from Pixiv, have an important metadata attached to them. For example - image titles and/or artist's commentary on Pixiv might describe the picture's story line, or deliver a punchline to a joke contained in the image.

**Describe the solution you'd like**

The only way of reliably attaching text information like this to an image, that I can come up with, is by writing it to the file's properties.
All the files regardless of their filetype have the following description properties:
Title
Subject
Rating
Tags
Comments
So, as I see it, the "Title" property could be used to store an individual image title, or a combination of a gallery name and page number if said image is part of a gallery.
"Comments" would then contain artist commentary.
"Tags" can obviously be used to store the image's tags, which could probably help organize local collection.
It would be somewhat counter-intuitive, but the "Subject" property could contain information about what pools or series the image is in?

**Describe alternatives you've considered**

Writing comments and titles to a text file with the same name as an image, maybe?

**Additional context**

None

**Comment by Bionus** [comment]
> All the files regardless of their filetype have the following description properties:

The problem is, they don't. PNG, GIF, and many other formats don't have those fields. And when they have, they are usually technically handled differently. There are not standards, which is why Grabber does not support it.

This is a request that often comes on the table, which is why [multiple issues already asked for it](https://github.com/Bionus/imgbrd-grabber/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+metadata). But the conclusion is always the same: if only a few formats support it, it's not really a good option (unless converting all images to JPG before saving them).

But if you know of a solution, I'd gladly implement it. I'm sure everyone would love being able to leverage the native search of the OS to find images based on their tags.

> Writing comments and titles to a text file with the same name as an image, maybe?

This was the alternative which was usually suggested in the issues on the subject, and is now 100% supported by Grabber (as long as the data you want is available in the tokens, which might not be the case for Pixiv artist comments, but that's another subject). Just create an "external log file" using the ".txt" suffix.

The suggestion usually is to use Hydrus in parallel with Grabber to handle your local collection, and have it read the text file you can generate with Grabber. A recommended format for the text file coudl be:

```
rating:%rating%
%all:includenamespace,excludenamespace=general,unsafe,spaces,separator=\n%
```

**Comment by RiabikovR** [comment]
I am currently using Directory Opus and it's metadata viewer appears to be able to write the "Comments" "Ratings" and "Tags" fields to any file (I don't know from where I took the "Subject" and "Title" fields, probably confused some image-specific metadata for file properties). Don't know, how well this would work outside of NTFS, tho.
Additionally, filesystem forks appear to be a step in the right general direction, even though their application and usefulness appear to vary greatly by filesystem.
https://en.wikipedia.org/wiki/Fork_(file_system)
As a last resort, I think there is ought to be some cross-platform C++ library to write image/video metadata.

**Comment by RiabikovR** [comment]
I can't seem to be able to upload an image for some reason, but here's a screenshot.
https://imgur.com/WqPvZUZ

**Comment by Bionus** [comment]
Seems like Directory Opus can read/write those, I'll have to see where it takes it from though (whether the actual ADS or a sidecar file like Hydrus/XMP). I think a few good solutions have been suggested recently so there might  be a way to implement this soon 😄 

Closing in favor of the most recent #2258 to have a single place of discussion
